executor/src/witgen/machines/mod.rs

```diff
@@ -57,17 +54,18 @@ pub trait Machine<'a, T: FieldElement>: Send + Sync {
     fn process_plookup<'b, Q: QueryCallback<T>>(
         &mut self,
         mutable_state: &'b mut MutableState<'a, 'b, T, Q>,
-        kind: IdentityKind,
-        left: &[AffineExpression<&'a AlgebraicReference, T>],
-        right: &'a SelectedExpressions<Expression<T>>,
-    ) -> Option<EvalResult<'a, T>>;
+        identity: IdentityId,
+        args: &[AffineExpression<&'a AlgebraicReference, T>],
+    ) -> EvalResult<'a, T>;
 
     /// Returns the final values of the witness columns.
     fn take_witness_col_values<'b, Q: QueryCallback<T>>(
         &mut self,
         fixed_lookup: &'b mut FixedLookup<T>,
         query_callback: &'b mut Q,
     ) -> HashMap<String, Vec<T>>;
+
+    fn identities(&self) -> Vec<IdentityId>;
```

Comment:
I think in future Rust versions we should be able to return `impl Iterator<Item = IdentityId>`, but this doesn't seem to be supported yet. It is only called rarely, so I think this is fine.